docs/source/working_with_aiida/caching.rst

```diff
@@ -65,88 +115,76 @@ This means that caching is enabled for ``TemplatereplacerCalculation`` and ``Str
-
-For calculations, the situation is a bit more complex: The node can have inputs and creates new data nodes as outputs. Again, the new node needs to keep its existing links. For the outputs, the calculation needs to create a copy of each node and link these as its outputs. This makes it look as if the calculation had produced these outputs itself, without caching.
-
-Finally, workchains can create links not only to nodes which they create themselves, but also to nodes created by a calculation that they called, or even their ancestors. This is where caching becomes impossible. Consider the following example (using workfunctions for simplicity):
-
-.. code:: python
+.. _caching_limitations:
 
-    from aiida.engine import workfunction
-    from aiida.orm import Int
+Limitations
+-----------
 
-    @workfunction
-    def select(a, b):
-        return b
+#. While caching saves unnecessary computations, the current implementation does not yet save disk space:
+   The output nodes of the cached calculation are full copies of the original outputs.
+   The plan is to add data deduplication as a global feature at the repository and database level (independent of caching).
 
-    d = Int(1)
-    r1 = select(d, d)
-    r2 = select(Int(1), Int(1))
+#. Workflow nodes are not cached. As explained below, in the current design this follows from the requirement that the provenance graph be independent of whether caching is enabled or not.
 
-The two ``select`` workfunctions have the same inputs as far as their hashes go. However, the first call uses the same input node twice, while the second one has two different inputs. If the second call should be cached from the first one, it is not clear which of the two input nodes should be returned.
+   * **Calculation nodes:** Calculation nodes can have data inputs and create new data nodes as outputs.
+     In order to make it look as if a cloned calculation produced its own outputs, the output nodes are copied and linked as well.
+   * **Workflow nodes:** Workflows differ from calculations in that they can *return* an input node or an output node created by a calculation.
+     Since caching does not care about the *identity* of input nodes but only their *content*, it is not straightforward to figure out which node to return in a cached workflow.
 
-While this example might seem contrived, the conclusion is valid more generally: Because workchains can return nodes from their history, they cannot be cached. Since even two equivalent workchains (with the same inputs) can have a different history, there is no way to deduce which links should be created on the new workchain without actually running it.
+  For the moment, this limitation is acceptable since the runtime of AiiDA WorkChains is usually dominated by expensive calculations, which are covered by the current caching mechanism.
 
```

Comment:
Remove the empty line of the usage of `#` will not work properly and the final item will start with `1.` again

Reply:
fixed. the line can actually stay - perhaps it was the wrong indentation.